base_layer/wallet/src/output_manager_service/storage/models.rs

```diff
@@ -87,6 +89,104 @@ impl DbWalletOutput {
+                }
+            } else {
+                let blocks_remaining = required_confirmations.saturating_sub(confirmations);
+                PayRefStatus::Pending(confirmations, blocks_remaining)
+            }
+        } else {
+            PayRefStatus::NotMined
+        }
+    }
+
+    /// Check if this output's PayRef matches the given reference
+    pub fn matches_payment_reference(&self, payref: &PaymentReference) -> bool {
+        if let Some(generated_payref) = self.generate_payment_reference() {
+            generated_payref == *payref
+        } else {
+            false
+        }
+    }
+
+    /// Get payment details for this output if it has a valid PayRef
+    pub fn get_payment_details(&self, current_tip_height: u64, required_confirmations: u64) -> Option<PaymentDetails> {
+        match self.get_payment_reference_status(current_tip_height, required_confirmations) {
+            PayRefStatus::Available(payref, confirmations) => Some(PaymentDetails {
+                payment_reference: payref,
+                commitment: self.commitment.clone(),
+                amount: self.wallet_output.value,
+                block_height: self.mined_height?,
+                block_hash: self.mined_in_block?,
+                mined_timestamp: self.mined_timestamp,
+                direction: self.infer_direction(),
+                status: self.status,
+                confirmations,
+                payment_id: Some(self.payment_id.to_bytes()),
+            }),
+            _ => None,
+        }
+    }
+
+    /// Infer the direction of this output based on its source
+    pub fn infer_direction(&self) -> PaymentDirection {
+        match self.source {
+            OutputSource::Coinbase => PaymentDirection::Received,
+            OutputSource::OneSided => PaymentDirection::Received,
+            OutputSource::StealthOneSided => PaymentDirection::Received,
+            OutputSource::HtlcRefund => PaymentDirection::Received,
+            OutputSource::AtomicSwap => PaymentDirection::Received,
+            OutputSource::Standard => {
+                // For standard outputs, distinguish between received payments and change outputs
+                if self.received_in_tx_id.is_some() {
+                    // Standard outputs with received_in_tx_id are typically change outputs from our own
+                    // transactions. This is because:
+                    // 1. Genuine received payments usually use OneSided or StealthOneSided sources
+                    // 2. Change outputs are created with Standard source during transaction building
+                    //
+                    // TODO: For better accuracy, we could check if received_in_tx_id corresponds to
+                    // an outbound transaction we initiated by querying the outbound_transactions table.
+                    PaymentDirection::SentChange
+                } else {
+                    PaymentDirection::Sent
+                }
```

Comment:
```suggestion
if self.payment_id == PaymetID::TransactionInfo{
PaymentDirection::SentChange}else {PaymentDirection::Sent}
```
This is not maybe, this is guaranteed to be accurate.